<a href="https://colab.research.google.com/github/Jamyly-NMelo/Universo-Aluna-Fiap/blob/main/Estat%C3%ADstica_com_Python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Estatística com Python

In [ ]:
import numpy as np
import sympy as smp
import pandas as pd
import seaborn as sns
import scipy

In [ ]:
import pandas as pd

In [ ]:
!pip install pandas=='0.22.0'

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.3/11.3 MB 24.7 MB/s eta 0:00:00
  error: subprocess-exited-with-error
  
  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  Installing build dependencies ... error
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.


# Introdução explicativa

# dados.csv

Chamaremos a biblioteca com pd seguido de read_csv() para lermos o arquivo dados.csv

In [ ]:
dados = pd.read_csv('dados.csv')

 head() do Pandas, visualizaremos somente os cinco primeiros registros de dados.




In [ ]:
dados.head()

UF  Sexo  Idade  Cor  Anos de Estudo  Renda    Altura
0  11     0     23    8              12    800  1.603808
1  11     1     23    2              12   1150  1.739790
2  11     1     35    8              15    880  1.760444
3  11     0     46    2               6   3500  1.783158
4  11     1     47    8               9    150  1.690631

 Anos de Estudo, conseguiremos fazer uma ordenação

In [ ]:
dados['Anos de Estudo']

0        12
1        12
2        15
3         6
4         9
         ..
76835    11
76836     7
76837    12
76838     4
76839     4
Name: Anos de Estudo, Length: 76840, dtype: int64

Para vermos apenas os valores únicos, aplicaremos .unique()

In [ ]:
dados['Anos de Estudo'].unique()

array([12, 15,  6,  9, 14, 16, 10,  1,  5,  7,  3,  2,  4,  8, 11, 17, 13])

O resultado será um array() fora de ordenação. Para ordenarmos os dados, adicionaremos a building function do Python chamada sorted()

In [ ]:
sorted(dados['Anos de Estudo'].unique())

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17]

 Usaremos o mesmo comando executado anteriormente, alterando apenas o conteúdo de dados[] com as variáveis corretas.

In [ ]:
sorted(dados['Sexo'].unique())

[0, 1]

In [ ]:
sorted(dados['Cor'].unique())

[0, 2, 4, 6, 8]

In [ ]:
sorted(dados['UF'].unique())

[11,
 12,
 13,
 14,
 15,
 16,
 17,
 21,
 22,
 23,
 24,
 25,
 26,
 27,
 28,
 29,
 31,
 32,
 33,
 35,
 41,
 42,
 43,
 50,
 51,
 52,
 53]

Escreveremos dados com Idade que pode ser aplicada em diversas operações, como min()

In [ ]:
dados.Idade.min()

13

poderemos exibir a idade máxima com max().

In [ ]:
dados.Idade.max()

99

Faremos uma impressão com print() recebendo 'De %s até %s anos' (%s são varieaveis) , seguido de % () contendo os dois últimos comandos anteriores.

In [ ]:
print('De %s até %s metros' % (dados.Altura.min(), dados.Altura.max()))

De 1.339244614 até 2.028496765 metros


alterando somente o nome da variável para Altura e a unidade de medida para metros dentro de print.

In [ ]:
print('De %s até %s metros' % (dados['Altura'].min(), dados['Altura'].max()))

De 1.339244614 até 2.028496765 metros


Se escrevermos ['Altura'] ao invés de somente Altura para pegarmos seus dados, o resultado será igual.



In [ ]:
print('De %s até %s anos' % (dados.Idade.min(), dados.Idade.max()))

De 13 até 99 anos


## Curso_de_Estatística_Parte_1.ipynb

Chamaremos os dados[] de 'Sexo' e faremos a contagem com o método .value_counts()

In [ ]:
dados['Sexo'].value_counts()


0    53250
1    23590
Name: Sexo, dtype: int64

*para a categoria 0 relativa ao sexo masculino, e 23590 para o feminino. *

Para fazermos isso, usaremos o mesmo método .value_counts() recebendo o parâmetro normalize sendo igual a True. Se ainda multiplicarmos por 100, veremos a porcentagem.

In [ ]:
dados['Sexo'].value_counts(normalize = True) * 100


0    69.299844
1    30.700156
Name: Sexo, dtype: float64

Para melhorarmos a apresentação dos dados, inseriremos o primeiro comando somente com .value_counts() sem parâmetro dentro de uma variável frequencia.

In [ ]:
frequencia = dados['Sexo'].value_counts()


In [ ]:
frequencia

0    53250
1    23590
Name: Sexo, dtype: int64

Faremos a mesma coisa com o comando que apresenta o percentual, e o colocaremos dentro da variável percentual na célula seguinte.

In [ ]:
percentual = dados['Sexo'].value_counts(normalize = True) * 100


In [ ]:
percentual

0    69.299844
1    30.700156
Name: Sexo, dtype: float64

Então, chamaremos de dist_freq_qualitativas sendo igual a pd.DataFrame() para o criarmos, onde passaremos um dicionário Python com {}

In [ ]:
dist_freq_qualitativas = pd.DataFrame({'Frequência': frequencia, 'Porcentagem (%)': percentual})


In [ ]:
dist_freq_qualitativas

Frequência  Porcentagem (%)
0       53250        69.299844
1       23590        30.700156

Na célula seguinte, chamaremos dist_freq_qualitativas com .rename() para renomearmos o index como um dicionário também, dizendo que 0 corresponde a 'Masculino' e 1 a 'Feminino'.

In [ ]:
dist_freq_qualitativas.rename(index = {0: 'Masculino', 1: 'Feminino'})


Frequência  Porcentagem (%)
Masculino       53250        69.299844
Feminino        23590        30.700156

Para sobrescrevermos e salvarmos as mudanças no arquivo, deveremos inserir o parâmetro inplace sendo igual a True no comando anterior.

In [ ]:
dist_freq_qualitativas.rename(index = {0: 'Masculino', 1: 'Feminino'}, inplace = True)


In [ ]:
dist_freq_qualitativas

Frequência  Porcentagem (%)
Masculino       53250        69.299844
Feminino        23590        30.700156

Não poderemos esquecer de colocar inplace igual a True para salvarmos as alterações.

In [ ]:
dist_freq_qualitativas.rename(index = {0: 'Masculino', 1: 'Feminino'}, inplace = True)
dist_freq_qualitativas.rename_axis('Sexo', axis = 'columns', inplace = True)


In [ ]:
dist_freq_qualitativas

Sexo       Frequência  Porcentagem (%)
Masculino       53250        69.299844
Feminino        23590        30.700156

Exercicios

In [107]:
dados = pd.DataFrame({'Profissão': [1, 2, 3, 1, 2, 2, 2, 3, 3, 2, 1, 3]})

In [108]:
dados['Profissão'].unique()

array([1, 2, 3])

In [109]:
sorted(dados['Profissão'].unique())

[1, 2, 3]

In [110]:
dados.Profissão.min()

1

In [111]:
dados.Profissão.max()

3

In [117]:
print('De %s até %s Profissão' % (dados.Profissão.min(), dados.Profissão.max()))

De 1 até 3 Profissão


In [119]:
frequencia = dados['Profissão'].value_counts()


In [115]:
dados['Frequência'].value_counts(normalize = True) * 100


2    41.666667
3    33.333333
1    25.000000
Name: Profissão, dtype: float64

In [116]:
sorted(dados['Profissão'].value_counts(normalize = True) * 100)


[25.0, 33.33333333333333, 41.66666666666667]

In [121]:
frequencia = dados['Profissão'].value_counts()

In [122]:
frequencia

2    5
3    4
1    3
Name: Profissão, dtype: int64

In [123]:
percentual = dados['Profissão'].value_counts(normalize = True) * 100


In [124]:
percentual

2    41.666667
3    33.333333
1    25.000000
Name: Profissão, dtype: float64

In [126]:
dist_freq_qualitativas = pd.DataFrame({'Profissão': frequencia, 'Porcentagem (%)': percentual})


In [127]:
dist_freq_qualitativas

Profissão  Porcentagem (%)
2          5        41.666667
3          4        33.333333
1          3        25.000000

In [128]:
dist_freq_qualitativas.rename(index = {1: 'Estatístico', 2: 'Cientista de Dados', 3: 'Programador Python'})


Profissão  Porcentagem (%)
Cientista de Dados          5        41.666667
Programador Python          4        33.333333
Estatístico                 3        25.000000

Acertei a pratica errei as perguntas